docs/admin/ServerSideKeygen.adoc

```diff
@@ -0,0 +1,157 @@
+
+[literal]
+policyset.userCertSet.3.constraint.class_id=keyConstraintImpl
+policyset.userCertSet.3.constraint.name=Key Constraint
+policyset.userCertSet.3.constraint.params.keyType=-
+policyset.userCertSet.3.constraint.params.keyParameters=1024,2048,3072,4096,nistp256,nistp384,nistp521
+policyset.userCertSet.3.default.class_id=serverKeygenUserKeyDefaultImpl
+policyset.userCertSet.3.default.name=Server-Side Keygen Default
+policyset.userCertSet.3.default.params.keyType=RSA
+policyset.userCertSet.3.default.params.keySize=2048
+policyset.userCertSet.3.default.params.enableArchival=true
+
+=== Authentication
+
+The two default server-side keygen enrollment profiles different in the authentication mechanism, where
+
+caServerKeygen_UserCert.cfg ::
+        contains empty value to *"auth.class_id="*, meaning that enrollment requests through this profile will require approval from a CA agent.
+caServerKeygen_DirUserCert.cfg ::
+        contains *"auth.instance_id=UserDirEnrollment"*, meaning that the user is required to pass LDAP uid/password authentication; Such authentication mechanism is considered as an automatic certificate issuance as it does not require per-request approval from a CA agent.
+
+Automatic approval could be configured by setting the auth.instance_id directive to any compatible authentication plugin class, as examplified in the caServerKeygen_DirUserCert.cfg profile mentioned above. Here is an example of such configuration in CS.cfg: 
+
+[literal]
+auths.instance.UserDirEnrollment.dnpattern=
+auths.instance.UserDirEnrollment.ldap.basedn=ou=People,dc=example,dc=com
+auths.instance.UserDirEnrollment.ldap.ldapconn.host=host.example.com
+auths.instance.UserDirEnrollment.ldap.ldapconn.port=389
+auths.instance.UserDirEnrollment.ldap.ldapconn.secureConn=false
+auths.instance.UserDirEnrollment.ldap.maxConns=
+auths.instance.UserDirEnrollment.ldap.minConns=
+auths.instance.UserDirEnrollment.ldapByteAttributes=
+auths.instance.UserDirEnrollment.ldapStringAttributes=mail
+auths.instance.UserDirEnrollment.pluginName=UidPwdDirAuth
+
+== Enrolling a Certificate Using Server-Side Keygen
+
+The default Sever-Side Keygen enrollment profile can be found on the EE page, under the “List Certificate Profiles” tab:
+
+=== Manual User Dual-Use Certificate Enrollment Using server-side Key generation
+
+.Server-Side Keygen Enrollment that requires agent manual approval
+image::Server-SideKeygenEnroll_manual.png[]
+
+=== Directory-authenticated User Dual-Use Certificate Enrollment Using server-side Key generation
+
+.Server-Side Keygen Enrollment that will be automatically approved upon successful LDAP uid/pwd authentication
+image::Server-SideKeygen_LDAP_auth.png[]
+
+Regardless of how the request is approved, the Server-Side Keygen Enrollment mechanism requires the End Entity user to enter a password for the PKCS12 package which will contain the issued certificate as well as the encrypted private key generated by the server once issued.
```

Comment:
In this doc the PKCS#12 sometimes is written as PKCS#12, PKCS12, or p12. I think it should be more consistent.